lib/index.js

```diff
@@ -63,28 +64,52 @@ resolver.query('example.com', 'A')
    * @param qtype {string} the type of record we're looking for (e.g. A, AAAA, TXT, MX)
    * @param method {string} Must be either "GET" or "POST"
    * @throws {MethodNotAllowedError} If the method is not allowed (i.e. if it's not "GET" or "POST"), a MethodNotAllowedError will be thrown.
-   * @returns {Promise<dnsPacket>} The DNS response received
+   * @returns {Promise<object>} The DNS response received
    */
   query(qname, qtype='A', method='POST') {
     return new Promise((resolve, reject) => {
       if (!isMethodAllowed(method)) {
         return reject(new MethodNotAllowedError(`Request method ${method} not allowed. Must be either 'GET' or 'POST'`))
       }
       let dnsMessage = makeQuery(qname, qtype);
-      sendDohMsg(dnsMessage, this.nameserver_url, method)
+      sendDohMsg(dnsMessage, this.nameserver_url, method, null)
+        .then(resolve)
+        .catch(reject)
+    });
+  }
+
+  /**
+   * This method is identical to {@link DohResolver.query}, except this method allows you to pass in custom HTTP headers.
+   * <br>
+   * <b><i>IMPORTANT: If you don't provide the "Accept: application/dns-message" header, you probably won't get the response you're hoping for.
+   * See [RFC 8484 examples](https://tools.ietf.org/html/rfc8484#section-4.1.1) for examples of HTTPS headers for both GET and POST requests.</i></b>
+   * @param qname {string} the domain name to query for (e.g. example.com)
+   * @param qtype {string} the type of record we're looking for (e.g. A, AAAA, TXT, MX)
+   * @param method {string} Must be either "GET" or "POST"
+   * @param headers {object} Custom HTTP headers to use in the DNS query
+   * @throws {MethodNotAllowedError} If the method is not allowed (i.e. if it's not "GET" or "POST"), a MethodNotAllowedError will be thrown.
+   * @returns {Promise<object>} The DNS response received
+   */
```

Comment:
Can it just be a single method with either a “additonalHeaders” field (with needed headers set) or a “headers” field with a default value that works.

Reply:
Good idea!